src/util/tools.cc

```diff
@@ -426,28 +431,29 @@ std::string dictEncodeSimple(const std::map<std::string, std::string>& dict)
     return dictEncode(dict, '/', '/');
 }
 
-void dictDecode(std::string_view url, std::map<std::string, std::string>& dict, bool unEscape)
+std::map<std::string, std::string> dictDecode(std::string_view url, bool unEscape)
 {
-    auto data = url.data();
-    auto dataEnd = data + url.length();
+    std::map<std::string, std::string> dict;
```

Comment:
The point that worries me is that the original version just added to the dict while now we create a **new** and reassign it in the calling class.

The is no problem for temporary objects, but changes logic for a second call with the same instance of a class like`WebRequestHandler`

Reply:
I am also a little unsure about this point. I have so far not really tested it - this is among other things the reasing this PR is still a draft. I will definitely look into it more closely when I find the time - and before setting this PR ready to merge.

Reply:
I rewrote all cases that would have changed due to the new `dictDecode` function. `CdsResource` now decodes directly in the constructor, so we can never overwrite existing values. And for `WebRequestHandler`, I added a `dictMerge` function that merges the existing `params` with new ones (even though I have not observed any non-empty `params` so far). `dictMerge` does `emplace`, which is what `dictDecode` did before.